sketcherMinimizerFragment.cpp

```diff
@@ -407,6 +407,27 @@ unsigned int sketcherMinimizerFragment::countHeavyAtoms() const
     return n;
 }
 
+unsigned int sketcherMinimizerFragment::countConstrainedAtoms() const
+{
+    int n = 0;
+    for (auto atom : m_atoms) {
+        if (atom->constrained)
+            n++;
```

Comment:
prefer prefix increment.

Maybe consider using std::count_if? Your call as to readability. I think it's something like:

    auto is_constrained = [](Atom* a) {return a->constrained;};
    return std::count_if(m_atoms.begin(), m_atoms.end(), is_constrained);

Reply:
I'm really not a fan of count_if... I find it more readable the way it is, with prefix increment of course